cornice/tests/test_validation.py

```diff
@@ -94,6 +94,16 @@ def test_accept(self):
         response = app.get('/service3', headers={'Accept': 'text/*'})
         self.assertEqual(response.content_type, "text/json")
 
+        # test that using a callable to define what's accepted works as well
+        # now, the callable returns a scalar instead of a list
+        response = app.put('/service3', headers={'Accept': 'audio/*'},
+                           status=406)
+        error_description = response.json['errors'][0]['description']
+        self.assertIn('text/json', error_description)
+
+        response = app.put('/service3', headers={'Accept': 'text/*'})
+        self.assertEqual(response.content_type, "text/json")
```

Comment:
Will do. The lines have been copied from the other test cases which are overall somehow not in the best shape (see next comment of yours and my minor improvements in #351). Will be happy to improve some aspects of that slightly as we go when having the opportunity if you would appreciate that.